src/Lean/Elab/StructInst.lean

```diff
@@ -505,8 +505,12 @@ mutual
             return { ref, lhs := [FieldLHS.fieldName ref fieldName], val := val } :: fields
           match Lean.isSubobjectField? env s.structName fieldName with
           | some substructName =>
-            -- If one of the sources has the subobject field, use it
-            if let some val ← s.source.explicit.findSomeM? fun source => mkProjStx? source.stx source.structName fieldName then
+            -- If src is a term for a parent field and the field is that parent projection, use it
+            -- This is necessary as findField? `A `toA returns none hence so will mkProjStx?
```

Comment:
Please add comments here noting the performance issue and explaining how this fix resolves it. Reading the comments here you wouldn't know that there was a performance issue to be addressed!